bgpd/bgp_open.h

```diff
@@ -35,11 +35,16 @@ struct capability_mp_data {
 };
 
 struct graceful_restart_af {
-	afi_t afi;
-	safi_t safi;
+	uint16_t afi;
+	uint8_t safi;
```

Comment:
why ?

Reply:
According to graceful restart RFC 4724           
         +--------------------------------------------------+
         | Address Family Identifier (16 bits)              |
         +--------------------------------------------------+
         | Subsequent Address Family Identifier (8 bits)    |
         +--------------------------------------------------+

Wrong size is defined in current FRR:
(gdb) p/x sizeof(afi_t)
$2 = 0x4
(gdb) p/x sizeof(safi_t)
$3 = 0x4

This causes failure in compatibility testing with other switches.

Reply:
But this value is not what is being sent over line inside of the packet. FRR has special code which convert from aft_t, safi_t to a line format

Reply:
Do you mean the big/small endian convert or something else?  Which piece of convert code do you refer to?

Reply:
https://github.com/FRRouting/frr/blob/0688fd81d006cbf4a0aedd771d769a497514ef4b/lib/iana_afi.h#L24
https://github.com/FRRouting/frr/blob/0688fd81d006cbf4a0aedd771d769a497514ef4b/lib/iana_afi.h#L65

Reply:
Here we just copy fields from wire directly, without conversion, similar to capability_mp_data.  There were multiple problems fixed after the introduction of iana_xxx_t, such as 
https://github.com/FRRouting/frr/commit/a46a2e9b4e8de782ac07e01429a80ed7ec167dcb